data_model/src/trigger/set.rs

```diff
@@ -120,14 +120,65 @@ impl Set {
         self.ids.iter().map(|entry| entry.key().clone()).collect()
     }
 
+    /// Apply `f` to triggers that belong to the given [`DomainId`]
+    ///
+    /// Returns an empty list if [`Set`] doesn't contain any trigger belonging to [`DomainId`].
+    pub fn inspect_by_domain_id<F, R>(&self, domain_id: &DomainId, f: F) -> Vec<R>
+    where
+        F: Fn(&Id, &dyn ActionTrait) -> R,
+    {
+        self.ids
+            .iter()
+            .filter_map(|pair| {
+                let id = pair.key();
+                let trigger_domain_id = id.domain_id.as_ref()?;
+
+                if trigger_domain_id != domain_id {
+                    return None;
+                }
+
+                let event_type = pair.value();
+
+                let result = match event_type {
```

Comment:
I think we could get rid of `result` variable and wrap `match` in `Some` directly.